openqaoa/problems/problem.py

```diff
@@ -321,6 +336,10 @@ def __init__(self,
         self.A = A if A else 2 * distance_matrix.max()
         self.B = B
 
+        # set the metadata
+        self.metadata = {'name': 'tsp', 'n_cities': self.n_cities}
```

Comment:
do you mean why `n_cities` is by default in the metadata? 
Because, when the QUBO is created this metadata is passed as a parameter.

However, I am not sure that `n_cities` here makes sense because `n` will be equal to `n_cities` (and the same happens for all the other type of problems, with `n_items`, `n_nodes`, `n_numbers`, etc.). Should we remove these attributes from the metadata?